src/javascript/app/pages/user/account/authenticate.js

```diff
@@ -339,31 +374,54 @@ const Authenticate = (() => {
         processFiles(files);
     };
 
+    const showNimcDialog = () =>  new Promise(resolve => {
```

Comment:
```suggestion
    const showNIMCDialog = () =>  new Promise(resolve => {
```

Reply:
No need to return the wrapped promise. Since you can just return `Dialog.confirm` function call.